RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc

```diff
@@ -76,7 +76,10 @@ class PhotonIDValueMapProducer : public edm::stream::EDProducer<> {
   reco::PFCandidate::ParticleType
   candidatePdgId(const edm::Ptr<reco::Candidate> candidate, bool isAOD);
 
-  std::pair<float,float> getTrackDxyDz(const edm::Ptr<reco::Candidate> & candidate, const reco::Particle::Point & vtxpos, bool isAOD);
+  const reco::Track* getTrackPointer(const edm::Ptr<reco::Candidate> candidate, bool isAOD);
+  void getImpactParameters(const edm::Ptr<reco::Candidate> candidate,
+                           bool isAOD, const reco::Vertex pv, float &dxy, float &dz);
```

Comment:
please pass args by reference as in the removed method
```const edm::Ptr<reco::Candidate>& candidate```, ```const reco::Vertex& pv```

Reply:
Changes implemented here:
https://github.com/lsoffi/cmssw/blob/CMSSW_9_4_0_pre3_TnP/RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc